### ISSUE TITLE
Error building rdkafka

### ISSUE THREAD

**Issue by amartin-ampere** [iopsystems/rpc-perf#109]
rdkafka fails to build due to a mismatched types error. The twitter rpc-perf repository builds successfully, but this fork does not.

### Expected behavior

Successful build. 

### Actual behavior

error[E0308]: mismatched types
   --> /home/amartin/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rdkafka-0.25.0/src/config.rs:153:17
    |
150 |             rdsys::rd_kafka_conf_get(
    |             ------------------------ arguments to this function are incorrect
...
153 |                 buf.as_mut_ptr() as *mut i8,
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `*mut u8`, found `*mut i8`
    |
    = note: expected raw pointer `*mut u8`
               found raw pointer `*mut i8`
note: function defined here
   --> /home/amartin/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rdkafka-sys-3.0.0+1.6.0/src/bindings.rs:922:12
    |
922 |     pub fn rd_kafka_conf_get(
    |            ^^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0308`.
error: could not compile `rdkafka` (lib) due to previous error
warning: build failed, waiting for other jobs to finish...

### Steps to reproduce the behavior

Error occurs when building on my system.

**Comment by thinkingfish** [comment]
This is an upstream bug that on linux+arm it maps `c_char` to `u8` but on x86 or macOS it maps `c_char` to `i8`, see: https://github.com/rust-lang/rust/issues/60226#issuecomment-486095925

Since the problem is upstream either we can patch it based on cfg rules or submit a PR upstream:


(in `~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rdkafka-0.25.0/src/config.rs`)
```
149         let res = unsafe {
150             rdsys::rd_kafka_conf_get(
151                 self.ptr(),
152                 key_c.as_ptr(),
153                 buf.as_mut_ptr() as *mut i8,                                                                                                                                     154                 &mut size,
155             )
156         };
```
and (`~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rdkafka-sys-3.0.0+1.6.0/src/bindings.rs`)

```
 921 extern "C" {
 922     pub fn rd_kafka_conf_get(                                                                                                                                                    923         conf: *const rd_kafka_conf_t,
 924         name: *const c_char,
 925         dest: *mut c_char,
 926         dest_size: *mut size_t,
 927     ) -> rd_kafka_conf_res_t;
 928 }
 ```

**Comment by amartin-ampere** [comment]
Thank you for the insight! Would a patch such as the one below break any functionality?

`
149         let res = unsafe {
150             rdsys::rd_kafka_conf_get(
151                 self.ptr(),
152                 key_c.as_ptr(),
153                 buf.as_mut_ptr() **as *mut u8**,                                                                                                                                     154                 &mut size,
155             )
156         };
`

**Comment by thinkingfish** [comment]
Looks like it's fixed in the latest version of `rust-rdkafka`. We will update the dependency and it should be fixed without you needing to do anything.

**Comment by brayniac** [comment]
This is fixed with #110 which updated the rdkafka dependency. Should build and run under `aarch64` now. Thanks again for reporting!

**Comment by amartin-ampere** [comment]
Thanks for the quick patch! Happy Halloween 👻
